pkg/handlers/launch.go

```diff
@@ -214,6 +308,12 @@ func (h *launchHandler) buildEnvVars(payload *launchPayload) (map[string]string,
 }
 
 func (h *launchHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
```

Comment:
can be for the next PR, but `ServeHTTP` is getting a bit chunky now, would be worth breaking it up to make the code easier to follow :-)

Reply:
Absolutely. That thing is a beast 🤣